### PR TITLE
Rs sample mm

### PR DESCRIPTION
**Pull request by RohanShrothrium**

### CODE REVIEW

File: src/market/cancel.ts [hunk 1]
```diff
@@ -13,19 +14,78 @@ export abstract class OrderCanceler {
...
+                    tx.gasPrice = baseGasPrice;
+                }
+            }
+
+            console.time('Transaction Send Time');
+            const transaction = await signer.sendTransaction(tx);
+            console.timeEnd('Transaction Send Time');
+
+            console.time('Transaction Wait Time');
+            const receipt = await transaction.wait(1);
```
Comment (devblixt) [inline src/market/cancel.ts:81]: is 1 confirmation ok or should we add it as an option?

File: src/market/gtc.ts [hunk 2]
```diff
@@ -92,18 +93,79 @@ export abstract class GTC {
...
+                tx.gasPrice = baseGasPrice;
+            }
+        }
+
+        console.time('Transaction Send Time');
+        const transaction = await signer.sendTransaction(tx);
+        console.timeEnd('Transaction Send Time');
+
+        console.time('Transaction Wait Time');
+        const receipt = await transaction.wait(1);
```
Comment (devblixt) [inline src/market/gtc.ts:161]: same qn as before

File: src/market/gtc.ts [hunk 3]
```diff
@@ -138,18 +200,79 @@ async function estimateGasBuy(
...
+            from: address,
+            data,
+            ...(txOptions?.nonce !== undefined && { nonce: txOptions.nonce }),
+            ...(txOptions?.gasLimit && { gasLimit: txOptions.gasLimit }),
+            ...(txOptions?.gasPrice && { gasPrice: txOptions.gasPrice }),
+            ...(txOptions?.maxFeePerGas && { maxFeePerGas: txOptions.maxFeePerGas }),
+            ...(txOptions?.maxPriorityFeePerGas && { maxPriorityFeePerGas: txOptions.maxPriorityFeePerGas })
+        };
+
+        console.time('RPC Calls Time');
```
Comment (devblixt) [inline src/market/gtc.ts:237]: should the benchmarking console logs exist?

File: src/market/cancel.ts (hunk 1 above)
Comment (RohanShrothrium) [reply src/market/cancel.ts:81]: Default is 1, I will remove this

File: src/market/gtc.ts (hunk 3 above)
Comment (RohanShrothrium) [reply src/market/gtc.ts:237]: I am pushing without it rn

**Review by devblixt — APPROVED** [review]

**Pull request merged**